### ISSUE TITLE
Dead Space 3 Cheats corrupting save

### ISSUE THREAD

**Issue by Zak1sBak** [bucanero/apollo-ps3#139]
Anytime I apply cheats to my Dead Space 3 save, the game says the save file is corrupt. Is there anything I can do to fix this?

**Comment by bucanero** [comment]
which title-id? and which codes are you applying?

some games have checksum verification that needs to be updated or the game will reject the file.

**Comment by Zak1sBak** [comment]
The title-id is BLUS31053. I'm applying max resources for Tungsten, Semiconductor, Scrap Metal, Somatic Gel, Transoucer, and Ration Seals.

**Comment by bucanero** [comment]
Dead Space 3 `BLUS31053` requires a checksum fix. There are 3 codes at the bottom that need to be added to fix the save. Have you selected those? if you didn't apply, the save will become corrupt.

If you did select the checksum fix codes and it still fails, then upload 2 copies of the save, one before changes (a valid working save), and then the save after the change (not working).
I can take a look when I have a chance to see if the changes and checksums are applied OK or not.

**Comment by Zak1sBak** [comment]
Alright here are the saves. I think the 3 codes at the bottom are always selected when I edit them
[BLUS31053-DS3_SAVE_V01_SLOT_02_corrupted.zip](https://github.com/bucanero/apollo-ps3/files/13756650/BLUS31053-DS3_SAVE_V01_SLOT_02_corrupted.zip)
[BLUS31053-DS3_SAVE_V01_SLOT_02_normal.zip](https://github.com/bucanero/apollo-ps3/files/13756652/BLUS31053-DS3_SAVE_V01_SLOT_02_normal.zip)

**Comment by bucanero** [comment]
I've checked the files, and the checksum is updated correctly in the `corrupt` save, so probably the issue is related to the cheat codes.

I'd suggest to test one code at a time. Also you can test updating the checksum only first (the last 3 codes), and see if the game loads the save after applying only a checksum update.

You can also try using the PC app "Bruteforce Save data", it also has the same cheats for Dead Space 3, and you can fix the checksum with this tool: https://github.com/bucanero/save-decrypters/blob/master/deadspace-checksum-fixer/samples/DS3_Checksum_Fixer.exe

**Comment by bucanero** [comment]
one additional thing, I was testing with the original checksum Fixer tool and the result with your `normal` file do not match, so maybe the game version you have uses a different checksum hash.

**Comment by Zak1sBak** [comment]
The save corrupted when I selected all codes at the bottom. I unselected those and used the Max Tungsten cheat only. It didn't seem to do anything.

**Comment by bucanero** [comment]
> The save corrupted when I selected all codes at the bottom. I unselected those and used the Max Tungsten cheat only. It didn't seem to do anything.

I see. In that case then the checksum fix is wrong and not matching the one expected by the game. As mentioned before, the last option to try would be using "Bruteforce Savedata" app on your PC, along with the `DS3 Checksum Fixer` tool.

If that doesn't work then we don't have a proper way to adjust the save to make it valid after modding values.

**Comment by Zak1sBak** [comment]
Where can I download the latest version of the Bruteforce Savedata app?

**Comment by bucanero** [comment]
@Zak1sBak you can get the PS3 tool pack from Aldostools, the author of Bruteforce savedata, that includes many tools including BSD: https://www.psx-place.com/resources/ps3-tools-collection.594/
download mirrors:
- https://www.upload.ee/files/15129894/install_ps3tools_4.8.1.exe.html
- https://turbobit.net/p2jflrquirgs.html

older version:
https://ps3.brewology.com/downloads/download.php?id=13974&mcid=4

**Comment by Zak1sBak** [comment]
Each of the downloads say they have a virus

**Comment by Zak1sBak** [comment]
Alright, I got Bruteforce working, but I'm not very experienced in using it. Gonna mess around for now.

**Comment by bucanero** [comment]
> Alright, I got Bruteforce working, but I'm not very experienced in using it. Gonna mess around for now.

yes, BSD might be a bit complex on a first look, but it should let you decrypt your PS3 saves, apply patches to the save, and encrypt it back so you can load those saves on your console. I suggest asking any questions on [psx-place forum](https://www.psx-place.com/), the author @aldostools is always there ready to answer.

**Comment by bucanero** [comment]
No, you can try with Bruteforce Savedata tool on PC, but it looks like the custom checksum tool is not creating valid checksums. You should test but keep a save backup in case it corrupts the data.
